icu4c/source/test/intltest/numfmtst.cpp

```diff
@@ -10033,4 +10034,29 @@ void NumberFormatTest::Test13733_StrictAndLenient() {
     }
 }
 
+void NumberFormatTest::Test21232_ParseTimeout() {
+    IcuTestErrorCode status(*this, "Test21232_ParseTimeout");
+
+    DecimalFormat df(status);
+    if (status.errDataIfFailureAndReset()) {
+        return;
+    }
+
+    UnicodeString input = u"4444444444444444444444444444444444444444";
+    if (quick) {
+        for (int32_t i = 0; i < 5; i++) {
+            input.append(input);
+        }
+        assertEquals("Somewhat long input of digits", 1280, input.length());
+    } else {
+        for (int32_t i = 0; i < 12; i++) {
+            input.append(input);
+        }
+        assertEquals("Very long input of digits", 163840, input.length());
+    }
+    Formattable result;
+    df.parse(input, result, status);
+    // Should not hang
```

Comment:
How to verify 'Should not hang'? The issue from the ticket was a timeout (after 60 secs). Can the code really hang here or is the real error symptom that processing takes unexpectedly long? Would it make sense to measure the time and raise an issue after 60 secs of elapse time? Somewhat imprecise, of course, depends on machine and machine load.

Reply:
I'm not sure of the best way to enforce this.  In exhaustive mode, on my machine, the test takes ~30 seconds before the change, and ~1 second after the change.